### PR TITLE
Check if Loupe is set up when interacting with documents

### LINKED ISSUE #51: TableNotFoundException on fresh initialized Loupe-Instance
When you initalize a fresh loupe instance without adding any documents the `getDocument` method will throw a Doctrine TableNotFoundException. I would rather expect that "null" is returned.

Example code to reproduce the bug:

https://phpsandbox.io/e/x/ynntx?layout=EditorPreview&defaultPath=%2F&theme=dark&showExplorer=no&openedFiles=

```php
<?php

use Loupe\Loupe\Configuration;
use Loupe\Loupe\Loupe;
use Loupe\Loupe\LoupeFactory;

touch(__DIR__ . '/database.sqlite');

$configuration = Configuration::create()
                ->withPrimaryKey('id') // optional, by default it's 'id'
                ->withSearchableAttributes(['name', 'contact_name', 'contact_email', 'comment', 'notice']) // optional, by default it's ['*'] - everything is indexed
                ->withFilterableAttributes(['client_id', 'object_type', 'updated_at', 'created_at'])
                ->withSortableAttributes(['updated_at', 'name']);
$loupeFactory = new LoupeFactory();
$loupe = $loupeFactory->create(__DIR__ . '/database.sqlite', $configuration);


$loupe->getDocument("doc1");
```

### PR DESCRIPTION
**Pull request by DumbergerL**
Fix #51

### CODE REVIEW

File: src/Internal/Engine.php [hunk 1]
```diff
@@ -90,11 +90,21 @@ public function getConnection(): Connection
         return $this->connection;
     }
 
+    public function tableExists(string $tableName): bool
```
Comment (Toflar) [inline src/Internal/Engine.php:93]: Not needed.

File: src/Internal/Engine.php [hunk 2]
```diff
@@ -90,11 +90,21 @@ public function getConnection(): Connection
...
+        $schemaManager = $this->getConnection()->createSchemaManager();
+        return $schemaManager->tablesExist($tableName);
+    }
+
     /**
      * @return array<string, mixed>|null
      */
     public function getDocument(int|string $identifier): ?array
     {
+        if(!$this->tableExists(IndexInfo::TABLE_NAME_DOCUMENTS)){
```
Comment (Toflar) [inline src/Internal/Engine.php:104]: Use `IndexInfo::needsSetup()` here.

File: src/Internal/Index/Indexer.php [hunk 3]
```diff
@@ -92,6 +92,10 @@ public function addDocuments(array $documents): IndexResult
      */
     public function deleteDocuments(array $ids): self
     {
+        if(!$this->engine->tableExists(IndexInfo::TABLE_NAME_DOCUMENTS)){
```
Comment (Toflar) [inline src/Internal/Index/Indexer.php:95]: Use `IndexInfo::needsSetup()` here.

File: tests/Functional/DocumentHandlingTest.php [hunk 4]
```diff
@@ -47,4 +47,16 @@ public function testGetDocument(): void
 
         $this->assertNull($loupe->getDocument('foobar'));
     }
+
+    public function testGetDocument_WithoutIndexedDocuments(): void
```
Comment (Toflar) [inline tests/Functional/DocumentHandlingTest.php:51]:
```suggestion
    public function testGetDocumentWhenNotSetUpYet(): void
```

File: tests/Functional/IndexTest.php [hunk 5]
```diff
@@ -289,6 +289,21 @@ public function testDeleteDocument(): void
         ]);
     }
 
+    public function testDeleteDocument_WithoutIndexedDocuments(): void
```
Comment (Toflar) [inline tests/Functional/IndexTest.php:292]:
```suggestion
    public function testDeleteDocumentWhenNotSetUpYet(): void
```

**Review by Toflar — CHANGES_REQUESTED** [review]
Hey, thanks for the PR!

Regarding Coding Style: Please run `composer cs`, it will fix it all for you :)

File: tests/Functional/IndexTest.php [hunk 6]
```diff
@@ -289,6 +289,20 @@ public function testDeleteDocument(): void
...
+    public function testDeleteDocumentWhenNotSetUpYet(): void
+    {
+        $configuration = Configuration::create()
+            ->withSearchableAttributes(['title', 'overview'])
+            ->withSortableAttributes(['title'])
+        ;
+
+        $loupe = $this->createLoupe($configuration);
+
+        // Delete document and assert it's gone
```
Comment (Toflar) [inline tests/Functional/IndexTest.php:301]: This comment can be removed, it's one of those "does not add any value" comments :) Literally it says what the method calls already say as well.

**Review by Toflar — COMMENTED** [review]
Minor nitpick, then we're good to go :)

File: tests/Functional/IndexTest.php (hunk 6 above)
Comment (Toflar) [reply tests/Functional/IndexTest.php:301]: (can also fix this myself if you want me to :))

Comment (DumbergerL) [reply tests/Functional/IndexTest.php]: No problem. I fixed it quickly. Thank you for your great work! I have been looking for a searchengine implemented in PHP for some time.

**Pull request merged**